Library/Homebrew/cask/audit.rb

```diff
@@ -570,13 +570,19 @@ def audit_rosetta
       odebug "Auditing Rosetta 2 requirement"
 
       extract_artifacts do |artifacts, tmpdir|
+        is_container = artifacts.any? { |a| a.is_a?(Artifact::App) || a.is_a?(Artifact::Pkg) }
+
         artifacts.filter { |a| a.is_a?(Artifact::App) || a.is_a?(Artifact::Binary) }
                  .each do |artifact|
+          next if artifact.is_a?(Artifact::Binary) && is_container
```

Comment:
If the `binary` artifact is totally separate from the `app` artifact (I believe that's possible?), this will skip the artifact incorrectly, right? Could it be possible to only skip the `binary` if it's path is a subdirectory of the `app` artifact?

Reply:
The issue here is  that artifacts outside the app bundle are what we want to ignore.  There's probably a better long term solution in the actual artifact extraction as another follow-up so consider this a tactical fix to avoid breakage.

See below where we pick up a shell script in `Content/Resources` for Visual Studio Code.

```
/usr/bin/env cp -pR /private/tmp/homebrew-unpack20240627-62563-va2rp7/Visual\ Studio\ Code.app/. /private/tmp/cask-audit20240627-62563-mcimq2/Visual\ Studio\ Code.app
/usr/bin/env lipo -archs /private/tmp/cask-audit20240627-62563-mcimq2/Visual\ Studio\ Code.app/Contents/MacOS/Electron
/usr/bin/env lipo -archs /Applications/Visual\ Studio\ Code.app/Contents/Resources/app/bin/code

audit for visual-studio-code: failed
 - Failed to determine artifact architecture!
```

Reply:
I guess I must be misunderstanding something, but I'm going to resolve this so this can be merged.

My thinking is that the `code` shell script lives under the `Visual Studio Code.app` path, so we can skip checking `code` specifically because the app bundle supersedes it.

However, a case like this might be different:

```ruby
cask "foo" do
  # ...
  app "foo.app"
  binary "foo-cli"
  # ...
end
```

In this case, `foo-cli` is totally separate from `foo.app` so it seems like it should still be checked. Whereas below, `foo-cli` is a part of the app bundle so it doesn't have to be checked as well:

```ruby
cask "foo" do
  # ...
  app "foo.app"
  binary "#{appdir}/foo.app/Contents/Resources/app/bin/foo-cli"
  # ...
end
```

Reply:
Sorry, @Rylan12 I don't think I said what I wanted to clearly.  That's 100% on me so I'll try again -

You are right.

Scripts and the like inside the `.app` bundle are skipped because they don't have an arch that can be checked with `lipo`.  For now, we'll take an assumption that all the binaries are the appropriate arch.

I think when I said "the issue here is that artifacts outside the app bundle are what we want to ignore in this case" I was referring to the anything that isn't the actual application binary.  I just said bundle when I shouldn't have.

Lack of sleep and all that.

i hope that makes sense.  What you said above is correct and is how it should be.

Reply:
No worries, thanks for the clarification! Agreed that the PR should be merged and can be improved later if we end up needing it